packages/peer-exchange/src/waku_peer_exchange_discovery.ts

```diff
@@ -39,36 +43,81 @@ export class PeerExchangeDiscovery
   implements PeerDiscovery
 {
   private readonly components: PeerExchangeComponents;
+  private readonly peerExchange: WakuPeerExchange;
   private readonly options: Options;
   private isStarted: boolean;
+  private intervals: Map<PeerId, NodeJS.Timeout> = new Map();
 
   private readonly eventHandler = async (
     event: CustomEvent<PeerProtocolsChangeData>
   ): Promise<void> => {
-    const { protocols } = event.detail;
+    const { protocols, peerId } = event.detail;
     if (!protocols.includes(PeerExchangeCodec)) return;
```

Comment:
Also check if an interval already exists for this `peerId`, if not, don't create a new one.